pkg/codegen/docs/gen.go

```diff
@@ -193,26 +204,35 @@ func (ss nestedTypeUsageInfo) add(s string, input bool) {
 		return
 	}
 
-	ss[s] = appearsIn{
+	ss[s] = typeUsage{
 		Input:  input,
 		Output: !input,
 	}
 }
 
-type typeDetails struct {
-	outputType   bool
-	inputType    bool
-	functionType bool
+// contains returns true if the token already exists and matches the
+// input or output flag of the token.
+func (ss nestedTypeUsageInfo) contains(token string, input bool) bool {
+	a, ok := ss[token]
+	if !ok {
+		return false
+	}
+
+	if input && a.Input {
+		return true
+	} else if !input && a.Output {
+		return true
+	}
+	return false
```

Comment:
Nit: This is fine, but it could be simplified...

Something like:

```suggestion
	if !ok {
		return false
	}
	return (input && a.Input) || (!input && a.Output)
```

or even just:

```suggestion
	return ok && ((input && a.Input) || (!input && a.Output))
```